### PR TITLE
33 circuit breaker for auth

### PR DESCRIPTION
**Pull request by theurichde**

### CODE REVIEW

File: cmd/aws-signing-proxy/main.go [hunk 1]
```diff
@@ -18,7 +21,7 @@ import (
 type EnvConfig struct {
 	TargetUrl             string `split_words:"true"`
 	Port                  int    `default:"8080"`
-	HealthPort            int    `default:"8081"`
+	MgmtPort              int    `default:"8081"`
```
[the remark this answers is not in the document]
Comment (theurichde) [reply cmd/aws-signing-proxy/main.go:24]: This port is now used for gathering /metrics and /status/health - any suggestions for another name?

File: pkg/circuitbreaker/circuit_breaker.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+var (
+	cbStateGauge   = promauto.NewGaugeVec(prometheus.GaugeOpts{Name: "auth_circuit_breaker_state", Help: "State of the authorization circuit breaker"}, []string{"state", "name"})
+	cbCounterGauge = promauto.NewGaugeVec(prometheus.GaugeOpts{Name: "auth_circuit_breaker_count", Help: "Circuit breaker request count"}, []string{"type"})
+)
+
+func (cb *CircuitBreaker) Execute(req func() (interface{}, error)) (interface{}, error) {
+
+	r, err := cb.breaker.Execute(req)
```
Comment (stat1981) [inline pkg/circuitbreaker/circuit_breaker.go:70]: I think with the r variable the request is meaning, but for me it is to shorten. Maybe we should rename it to the full varibale name request?

File: pkg/vault/vault.go [hunk 3]
```diff
@@ -59,9 +60,15 @@ func (c *Client) ReadFrom(path string) *ReadClient {
 	return r
 }
 
+var breaker = circuitbreaker.NewCircuitBreaker("vault-circuit-breaker")
```
Comment (stat1981) [inline pkg/vault/vault.go:63]: Here the name of the circuit breaker is vaul-circuit-breaker, but for OICD only oicd, there the naming pattern is not quite consistent.

Comment (stat1981) [reply pkg/vault/vault.go:63]: Or better remove the name, because there is only one proxy type running.

**Review by stat1981 — APPROVED** [review]

File: go.mod [hunk 4]
```diff
@@ -1,16 +1,36 @@
 module github.com/idealo/aws-signing-proxy
 
-go 1.18
+go 1.19
 
 require (
-	github.com/aws/aws-sdk-go v1.44.131
-	github.com/go-co-op/gocron v1.17.1
+	github.com/aws/aws-sdk-go v1.44.105
```
[the remark this answers is not in the document]
Comment (theurichde) [reply go.mod:6]: Ah, messed up the correct main rebase. thx for pointing that out

File: cmd/aws-signing-proxy/main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (stat1981) [reply cmd/aws-signing-proxy/main.go:24]: Now that it's clearer why Managment Port, I would keep the name.

**Review by roechi — APPROVED** [review]

**Review by stat1981 — APPROVED** [review]

**Pull request merged**